File: Exercises/Exercise 5/Starter/shop/src/app/entities/Product.ts
import { Brand } from "./Brand";
import { ProductGroup } from "./ProductGroup";
import { Specification } from "./Specification";

export class Product
{
    public id:number;
    public brand:Brand;
    public name:string;
    public productGroup:ProductGroup;
    public score: number;
    public image:string;  
    public specifications?:Specification[]; 

    constructor()
    {
        this.id = 0;
        this.brand = new Brand();
        this.image = "";
        this.name = "";
        this.productGroup = new ProductGroup();
        this.score = 0;
    }
}

File: DataLayer/DataLayer.Interfaces/Entities/Product.cs
using System.Collections.Generic;

namespace DataLayer.Interfaces.Entities
{
    public class Product
    {
        public int ID { get; set; }
        public string? Name { get; set; }
        public Brand? Brand { get; set; }
        public ProductGroup? ProductGroup { get; set; }
        public decimal Price { get; set; }
        public decimal Score { get; set; }
        public string? Image { get; set; }
        public ICollection<Specification> Specifications { get; set; } = new HashSet<Specification>();
    }
}